src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/AbstractLanguageServer.cs

```diff
@@ -196,8 +196,14 @@ public DelegatingEntryPoint(string method, AbstractTypeRefResolver typeRefResolv
                 var handlerEntryPoints = new Dictionary<string, (MethodInfo, RequestHandlerMetadata)>();
                 foreach (var metadata in handlersForMethod)
                 {
-                    var requestType = _typeRefResolver.Resolve(metadata.RequestTypeRef.GetValueOrDefault()) ?? NoValue.Instance.GetType();
-                    var responseType = _typeRefResolver.Resolve(metadata.ResponseTypeRef.GetValueOrDefault()) ?? NoValue.Instance.GetType();
+                    var noValueType = NoValue.Instance.GetType();
+
+                    var requestType = metadata.RequestTypeRef is TypeRef requestTypeRef
```

Comment:
When it's null. This is just a null check. Previously, `TypeRef` was a struct and `RequestTypeRef` is defined as `TypeRef?`. So, when it was a struct, `is TypeRef` would pull out the value. When I changed `TypeRef` to a class, I decided to leave `TypeRef` here rather than switching over to `metadata.RequestTypeRef is { } requestTypeRef`, since the type name is pretty short.